Mastodon/Scene/HomeTimeline/HomeTimelineViewModel.swift

```diff
@@ -146,32 +146,67 @@ extension HomeTimelineViewModel {
 extension HomeTimelineViewModel {
 
     // load timeline gap
-    func loadMore(item: StatusItem) async {
+    @MainActor
```

Comment:
What's th reason to run this entire code on `Main`, including the download?

Reply:
The download is run off the mainThread because `dataTaskPublisher` will dispatch to a worker thread by default. Only the code inside this function is run explicitly on the main thread to ensure data consistency when modifying the status records.